src/optable.ts

```diff
@@ -501,6 +501,128 @@ optable[opcodes.DELETE_FAST] = function(f: Py_FrameObject) {
     var i = f.readArg();
     f.locals.del(f.codeObj.varnames[i]);
 }
+// Searches for an exception handler inside the current Py_FrameObject
+// Uses that blockstack to move forwards or backwards in code (by changing lastInst)
+// Returns true if found, else returns false
+function find_exception_handler(f: Py_FrameObject) : boolean {
+    while (f.blockStack.length > 0) {
+        var b = f.blockStack[f.blockStack.length - 1];
+        f.blockStack.pop();
+        if (b[3] === opcodes.SETUP_EXCEPT) {
+            setup_block(f, opcodes.EXCEPT_HANDLER);
+            var endPos: number = b[2];
+            f.lastInst = endPos;
+            return true;
+        }
+        if (b[3] === opcodes.SETUP_FINALLY) {
+            var endPos: number = b[2];
+            f.lastInst = endPos;
+            return true;
+        }
+    }
+    return false;
+}
+// Unpack the lnotab to extract instruction/line numbers
+function unpack(str: any) {
+    var ln_byte_tuple: [number,number][] = [];
+    for(var i = 0, n = str.length; i < n; i+=2) {
+        var char1:number = parseInt(str.charCodeAt(i));
+        var char2:number = parseInt(str.charCodeAt(i+1));
+        ln_byte_tuple.push([char1, char2]);
+    }
+    return ln_byte_tuple;
+}
+
+// Using lnotab, find the linenumber of the current instruction in the .py file
+function addr2line(f: Py_FrameObject) {
+    var lineno = 0;
+    var addr = 0;
+    var chars = f.codeObj.lnotab.toString();
+    var lnotab = unpack(chars);
+    for (var i = 0; i < lnotab.length; i++) {
+        addr += lnotab[i][0];
+        if (addr > f.lastInst) {
+            break;
+        }
+        lineno += lnotab[i][1];
+    }
+    return lineno;
+}
+// Add traceback
+function frame_add_traceback(f: Py_FrameObject, t:Thread) {
+
+    var current_line = (f.codeObj.firstlineno) + addr2line(f);
+    var tback: string = "";
+    tback = "";
+    tback += "  File \"" + f.codeObj.filename.toString() +
```

Comment:
You might find it easier to use TypeScript [template strings](http://basarat.gitbooks.io/typescript/content/docs/template-strings.html).